repair/repair_task.hh

```diff
@@ -57,24 +61,31 @@ private:
     dht::token_range_vector _ranges;
     std::unordered_map<dht::token_range, repair_neighbors> _neighbors;
     streaming::stream_reason _reason;
-    shared_ptr<node_ops_info> _ops_info;
+    optimized_optional<abort_source::subscription> _abort_subscription;
 public:
     data_sync_repair_task_impl(tasks::task_manager::module_ptr module, repair_uniq_id id, std::string keyspace, std::string type, std::string entity, dht::token_range_vector ranges, std::unordered_map<dht::token_range, repair_neighbors> neighbors, streaming::stream_reason reason, shared_ptr<node_ops_info> ops_info)
         : repair_task_impl(module, id.uuid(), id.id, std::move(keyspace), "", std::move(type), std::move(entity), tasks::task_id::create_null_id())
         , _ranges(std::move(ranges))
         , _neighbors(std::move(neighbors))
         , _reason(reason)
-        , _ops_info(ops_info) 
-    {}
+    {
+        if (ops_info && ops_info->as) {
+            _abort_subscription = ops_info->as->subscribe([this] () noexcept {
+                (void)abort();
+            });
+        }
+    }
+
+    virtual tasks::is_abortable is_abortable() const noexcept override {
+        return tasks::is_abortable::yes;
```

Comment:
I think this task should not be abortable.
We should abort using the top level tasks which today is `user_requested_repair_task_impl`
and in the future it would be a node_ops task, that would recursively abort either this task, with repair based node operations enabled, or a corresponding streaming task.

Reply:
Maybe return `tasks::is_abortable(!_abort_subscription)`